src/WebJobs.Script.WebHost/WebJobsServiceCollectionExtensions.cs

```diff
@@ -82,6 +85,10 @@ public static IServiceProvider AddWebJobsScriptHost(this IServiceCollection serv
             // The services below need to be scoped to a pseudo-tenant (warm/specialized environment)
             builder.Register<WebScriptHostManager>(c => c.Resolve<WebHostResolver>().GetWebScriptHostManager()).ExternallyOwned();
             builder.Register<ISecretManager>(c => c.Resolve<WebHostResolver>().GetSecretManager()).ExternallyOwned();
+            builder.RegisterType<WebFunctionsManager>().As<IWebFunctionsManager>().SingleInstance();
+            builder.RegisterType<VirtualFileSystem>();
+            builder.RegisterType<ZipFileSystem>();
+            builder.RegisterType<VirtualFileSystemMiddleware>();
```

Comment:
Was the middleware registration here required? THe call to UseMiddleware should be sufficient

Reply:
done